src/main/kotlin/com/atkinsondev/maestro/MaestroTest.kt

```diff
@@ -1,19 +1,25 @@
 package com.atkinsondev.maestro
 
 import org.gradle.api.DefaultTask
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.file.RegularFileProperty
 import org.gradle.api.provider.Property
 import org.gradle.api.tasks.Input
 import org.gradle.api.tasks.Optional
 import org.gradle.api.tasks.TaskAction
+import org.gradle.process.ExecOperations
 import java.io.File
+import javax.inject.Inject
 
-abstract class MaestroTest : DefaultTask() {
+abstract class MaestroTest @Inject constructor(
+    private val execOperations: ExecOperations,
+) : DefaultTask() {
     @get:Input
-    abstract val flowsDir: Property<File>
+    abstract val flowsDir: DirectoryProperty
 
     @get:Input
     @get:Optional
-    abstract val screenshotFlowFile: Property<File>
+    abstract val screenshotFlowFile: RegularFileProperty
```

Comment:
This is probably a breaking change for users of Gradle's Kotlin script support so if we wanna try to avoid that I can try to do something.